spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/elasticsearch/ElasticsearchRestClientAutoConfiguration.java

```diff
@@ -80,6 +84,19 @@ RestClientBuilder elasticsearchRestClientBuilder(ElasticsearchRestClientProperti
 			return builder;
 		}
 
+		@Bean
+		@Conditional(IsSnifferAvailableOnClasspathCondition.class)
+		@ConditionalOnMissingBean(RestHighLevelClient.class)
+		Sniffer elasticsearchSnifferBuilder(ElasticsearchRestClientProperties properties,
+											RestClient elasticSearchRestClient) {
```

Comment:
Have you run the test before pushing an update? As I've explained several times in this issue, we do not expose a RestClient so requiring a RestClient bean is never going to work.